src/coreclr/src/vm/methodtablebuilder.cpp

```diff
@@ -11605,17 +11605,20 @@ BOOL MethodTableBuilder::ChangesImplementationOfVirtualSlot(SLOT_INDEX idx)
 
         if (!fChangesImplementation)
         {
-            // Search the previous slots in the parent vtable for the same implementation. If it exists and it was
-            // overriden, the ClassLoader::PropagateCovariantReturnMethodImplSlots will propagate the change to the current
-            // slot (idx), so the implementation of it will change.
-
             MethodDesc* pParentMD = ParentImpl.GetMethodDesc();
-            for (SLOT_INDEX i = 0; i < idx; i++)
+            if (pParentMD->RequiresCovariantReturnTypeChecking())
```

Comment:
This isn't correct. The propagation feature isn't actually tied to covariant return type checking, and I probably should have caught that in the previous change, as the PropagateCovariantReturnMethodImplSlots name isn't really right.